automated/linux/block-device/block-device.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0-only
+# Copyright (C) 2019 Linaro Ltd.
+
+# shellcheck disable=SC1091
+. ../../lib/sh-test-lib
+OUTPUT="$(pwd)/output"
+RESULT_FILE="${OUTPUT}/result.txt"
+
+PARTDEV=""
+PARTNAME=""
+DEVICE=""
+DEVNAME=""
+
+SKIP_INSTALL="false"
+FORMAT_DEVICE="false"
+
+MOUNT_DIR=$(mktemp -d "/tmp/block-device.XXXXX")
+TMPFILE=$(mktemp "/tmp/10M.XXXXX")
+RESULT=pass
+
+usage() {
+	echo "\
+	Usage: $0
+		     [-d </dev/sdXN>]
+		     [-s  <true|false>]
+		     [-f ]
+
+	-d </dev/sdXN>
+		This is the block device to be tested, eg. /dev/sda1.
+		It should be a partition, not a raw block device
+	-f
+		This will erase the device and create a new partition table
+		with a single ext4 partition on it.
+	-s <true|false>
+		Tell the test to skip installation of dependencies, or not
+
+	This test will perform tests on a block device.
+	"
+}
+
+while getopts "d:fs:h" opts; do
+	case "$opts" in
+		d) PARTDEV="${OPTARG}";
+		   PARTNAME=$(basename -- "${PARTDEV}");
+		   DEVICE=${PARTDEV/%[[:digit:]]/}
+		   DEVNAME=$(basename -- "${DEVICE}")
+		   ;;
+		f) FORMAT_DEVICE="true" ;;
+		s) SKIP_INSTALL="${OPTARG}" ;;
+		h|*) usage ; exit 1 ;;
+	esac
+done
+
+install() {
+	if [ "${SKIP_INSTALL}" = "true" ] || [ "${SKIP_INSTALL}" = "True" ]; then
+		info_msg "Skip installing dependencies"
+	else
+		# install dependencies
+		dist=
+		dist_name
+		case "${dist}" in
+			debian|ubuntu)
+				pkgs="bonnie++"
+				install_deps "${pkgs}" "${SKIP_INSTALL}"
+				;;
+			fedora|centos)
+				pkgs="bonnie++"
+				install_deps "${pkgs}" "${SKIP_INSTALL}"
+				;;
+			# When we don't have a package manager
+			# Assume dependencies pre-installed
+			*)
+				echo "Unsupported distro: ${dist}! Package installation skipped!"
+				;;
+		esac
+	fi
+}
+
+block_device_exists () {
+	local device=$1
+
+	if [[ -b ${device} ]]; then
+		echo true
+	else
+		echo false
+	fi
+}
+
+wait_for_device () {
+	local device=$1
+        local retries=20
```

Comment:
this indentation uses spaces and all other indents use tabs.

Reply:
Ooops!